incubator/hnc/internal/reconcilers/object_test.go

```diff
@@ -74,6 +75,18 @@ var _ = Describe("Exceptions", func() {
 			treeSelector: c2,
 			want:         []string{},
 			notWant:      []string{c1, c2, c3},
+		}, {
+			name:         "not propagate to any object when noneSelector is set to true",
+			noneSelector: "true",
```

Comment:
Try an explicit `false` and a bad value too? Actually all of them should have bad values to show that they're ignored.

Reply:
done. Also added bad values for selector and treeselector

Reply:
Ah I just realized that we are not currently checking the validity of selector and treeSelector. For example if we put a random string to either of the two selectors, it will turn out not to match anything. Maybe this is to be done with the webhook?

Reply:
You don't need to improve the testing of selector and treeSelector in this PR, but you should add tests for them. We need to know what will happen (and document it) if the user gets past the webhook. Webhooks are for usability but we never assume they're there.

Reply:
The current behavior is that invalid selector value will result in this object not propagating to any ns.

Do you think we need to check and ignore any selector value that doesn't map to the child namespaces?

Reply:
I document the current behavior before `getSelectorFromString()`

Reply:
lgtm